rclcpp/test/rclcpp/test_node.cpp

```diff
@@ -2733,13 +2733,33 @@ TEST_F(TestNode, get_publishers_subscriptions_info_by_topic) {
 
 TEST_F(TestNode, callback_groups) {
   auto node = std::make_shared<rclcpp::Node>("node", "ns");
-  size_t num_callback_groups_in_basic_node = node->get_callback_groups().size();
```

Comment:
we could still use `get_callback_groups().size()` here.
If we want to delete `get_callback_groups()`, which is a good idea IMO, we could add a `callback_groups_size()` method.

Reply:
Yeah, we could.  Right now we only seem to get the `size` of the callback group in tests, so I'm OK with the more verbose way of doing it.  In the future if we decide we need it in more places, we can add it in.

Reply:
I also think it would be a good idea to get rid of `get_callback_groups()` in its current form, because a) it would force users of it that are outside of our code base to consider if they're using it in a thread-unsafe way and b) because we could provide a const method that returns a copy of the callback groups vector instead, which could be used for `size()` and other things, which I would prefer over a dedicated `get_callback_groups_size()`.